salt/_auth/kubernetes_rbac.py

```diff
@@ -116,7 +151,34 @@ def _groups_basic(kubeconfig, username, token):
 
 @_log_exceptions
 def _auth_bearer(kubeconfig, username, token):
-    return _check_k8s_creds(kubeconfig, 'Bearer {}'.format(token))
+    """Check the provided bearer token using the TokenReview API."""
+    client = kubernetes.client.ApiClient(configuration=kubeconfig)
+    authn_api = kubernetes.client.AuthenticationV1Api(api_client=client)
+
+    token_review = authn_api.create_token_review(
+        body=kubernetes.client.V1TokenReview(
+            spec=kubernetes.client.V1TokenReviewSpec(token=token)
+        )
+    )
+
+    if token_review.status.error:
+        log.error('Failed to create TokenReview: %s',
+                  token_review.status.error)
+        return False
+
+    if token_review.status.authenticated:
+        if token_review.status.user.username != username:
+            log.error(
+                "Provided token belongs to '%s', does not match '%s'",
+                token_review.status.user.username,
+                username,
+            )
+            return False
+        else:
+            return True
+    else:
+        log.error('Provided token failed to authenticate')
```

Comment:
Can we log the provided `username` here?

Reply:
Done